src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -367,6 +388,175 @@ ErrorIfModifyQueryNotSupported(Query *queryTree)
 
 
 /*
+ * Fails if the expression contains STABLE functions which accept any paramaters derived
+ * from a Var. Assumes the expression contains no VOLATILE functions.
+ *
+ * Var's are allowed, but only if they are passed solely to IMMUTABLE functions
+ */
+static bool ContainsDisallowedFunctionCalls(Node *expression)
+{
+	bool unused = false;
+	return ContainsDisallowedFunctionCallsWalker(expression, &unused);
+}
+
+
+static bool ContainsDisallowedFunctionCallsWalker(Node *expression, bool *containsVar)
```

Comment:
When this has come up before in other projects I've added an Assert() that we're running one of a whitelist of versions, with a comment explaining what could go wrong with a newer version. That way the first test run during a port to a newer version will helpfully fail. Is that kosher here?